mozilla-central/setup

```diff
@@ -153,48 +183,69 @@ if [ -n "$INDEXED_GIT_REV" ]; then
-        # See bug 1487583 comment 3 for some more discussion.
         pushd generated-$PLATFORM
-        set +x  # Turn off echoing of commands and output only relevant things to avoid bloating logfiles
-        find . -type f |
-        while read GENERATED_FILE; do
-            if [ ! -f "../objdir/$GENERATED_FILE" ]; then
-                # Generated file wasn't encountered for previously processed platforms
-                echo "Taking generated file $GENERATED_FILE from $PLATFORM"
-                move_file "$GENERATED_FILE" "../objdir/$GENERATED_FILE"
-                if [ -f "../analysis-$PLATFORM/__GENERATED__/$GENERATED_FILE" ]; then
-                    # Move the analysis file as well
-                    move_file "../analysis-$PLATFORM/__GENERATED__/$GENERATED_FILE" "../analysis/__GENERATED__/$GENERATED_FILE"
-                fi
-            fi
-        done
-        set -x
+        find . -type f >> ../generated-files.list
         popd
 
-        # Throw away any leftover generated files and their analysis, so that they don't
-        # accidentally get merged later. The above loop should have copied any that we
-        # care about into the main objdir/ and analysis/ folders.
-        rm -rf generated-$PLATFORM
-        rm -rf analysis-$PLATFORM/__GENERATED__
-
         # List all the analysis files we have left. We will merge these across platforms
-        # after this per-platform loop is complete.
+        # after this per-platform loop is complete. Make sure to skip over the __GENERATED__
+        # directory
         pushd analysis-$PLATFORM
-        find . -type d >> ../analysis-dirs.list
-        find . -type f >> ../analysis-files.list
+        find . -not \( -name __GENERATED__ -prune \) -type d >> ../analysis-dirs.list
+        find . -not \( -name __GENERATED__ -prune \) -type f >> ../analysis-files.list
         popd
 
     done    # end PLATFORM loop
 
     date
 
+    # Special case: xptdata.cpp is a giant file and is different for each platform, but
+    # the differences are not particularly relevant so let's just keep the linux one.
+    rm -f              generated-macosx64/xpcom/reflect/xptinfo/xptdata.cpp
+    rm -f                 generated-win64/xpcom/reflect/xptinfo/xptdata.cpp
+    rm -f analysis-macosx64/__GENERATED__/xpcom/reflect/xptinfo/xptdata.cpp
+    rm -f    analysis-win64/__GENERATED__/xpcom/reflect/xptinfo/xptdata.cpp
+
+    # For each generated file, if all platforms generated the same thing (or didn't
+    # generate the file at all), copy it to the merged objdir.
+    set +x  # Turn off echoing of commands and output only relevant things to avoid bloating logfiles
+    sort generated-files.list | uniq |
+    while read GENERATED_FILE; do
+        ALL_SAME_AS=$(check_all_same generated-*/$GENERATED_FILE)
+        if [ "$ALL_SAME_AS" != "" ]; then
+            echo "Generated file $GENERATED_FILE was identical across platforms"
```

Comment:
Maybe this should end with "across platforms where it was created" to spell out the possibility that it existed on some platforms and not others?